templates/task-buildah.yaml

```diff
@@ -0,0 +1,125 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: buildah
+  labels:
+    app.kubernetes.io/version: {{ .Chart.Version }}
+{{- if .Values.annotations }}
+  annotations:
+  {{- .Values.annotations | toYaml | nindent 4 }}
+{{- end }}
+spec:
+  description: |
+    Buildah task builds source into a container image and
+    then pushes it to a container registry.
+  params:
+    - name: IMAGE
+      type: string
+      description: |
+        Reference of the image buildah will produce.
+    - name: REGISTRY
+      type: string
+      description: |
+        Full qualified registry url, including image name
+    - name: STORAGE_DRIVER
+      type: string
+      default: overlay
+      description: |
+        Set buildah storage driver
+    - name: CONTAINERFILE_PATH
+      type: string
+      default: ./Containerfile
+      description: |
+        Path to the `Containerfile` (`Dockerfile`) to build.
+    - name: CONTEXT_SUBDIRECTORY
+      type: string
+      default: .
+      description: |
+        Relative subdirectory to the `source` Workspace used as "context".
+    - name: TLS_VERIFY
+      type: string
+      default: "true"
+      description: |
+        Sets the TLS verification flags, `true` is recommended.
+    - name: BUILD_EXTRA_ARGS
+      type: string
+      default: ""
+      description: |
+        Extra parameters passed for the build command when building images.
+    - name: PUSH_EXTRA_ARGS
+      type: string
+      default: ""
+      description: |
+        Extra parameters passed for the push command when pushing images.
+    - name: SKIP_PUSH
+      default: "false"
+      description: |
+        Skip pushing the built image
+    - name: VERBOSE
+      type: string
+      default: "false"
+      description:
+        Turns on verbose logging, all commands executed will be printed out.
+
+  workspaces:
+    - name: source
+      optional: true
```

Comment:
I don't think this needs to be an optional workspace (otherwise, there is nothing to build)

Reply:
We'll follow-up on this 👼🏼